### PR TITLE
Add new BBEs for sync send and async send action

### PR DESCRIPTION
**Pull request by prakanth97**
## Purpose
Add following BBEs,

- Sync send action
- Async send action

### CODE REVIEW

File: examples/async-send/async_send.metatags [hunk 1]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates how async send action can be used for concurrency in Ballerina.
+keywords: ballerina, ballerina by example, bbe, async send, concurrency
```
Comment (SasinduDilshara) [inline examples/async-send/async_send.metatags:2]: Can't we add `worker`

File: examples/index.json [hunk 2]
```diff
@@ -889,6 +889,20 @@
                 "verifyBuild": true,
                 "verifyOutput": true,
                 "isLearnByExample": true
+            },
+            {
+                "name": "Sync-send",
```
Comment (SasinduDilshara) [inline examples/index.json:894]: Isn't it better something like `Synchronization Message passing`
WDYT?

File: examples/index.json [hunk 3]
```diff
@@ -889,6 +889,20 @@
...
+                "url": "sync-send",
+                "verifyBuild": true,
+                "verifyOutput": true,
+                "isLearnByExample": true
+            },
+            {
+                "name": "Async-send",
```
Comment (SasinduDilshara) [inline examples/index.json:901]: Isn't it better something like `Asynchronize Message passing`
WDYT?

File: examples/sync-send/sync_send.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Sync send
+
+Use `->> W` to send a message synchronously to worker `W`. These messages should be of `anydata` type. Unlike asynchronous send, synchronous send will wait until the message is delivered. Synchronous send will return nil if the message was successful to the receiver or will return an error or panic based on the receiving worker’s state.
+
```
Comment (SasinduDilshara) [inline examples/sync-send/sync_send.md:4]: What you mean by `return panic`?.

File: examples/async-send/async_send.bal [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import ballerina/io;
+
+public function main() {
+    worker A {
+        int num = 10;
+
+        // Sends integer value `10` to worker `B` asynchronously.
```
Comment (praneesha) [inline examples/async-send/async_send.bal:7]:
```suggestion
        // Sends the `10` integer value to the `B`  worker asynchronously.
```

File: examples/async-send/async_send.bal [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        num -> B;
+        
+        // Receives string `Hello` from worker `B`.
```
Comment (praneesha) [inline examples/async-send/async_send.bal:10]:
```suggestion
        // Receives the `Hello` string from the `B` worker.
```

File: examples/async-send/async_send.bal [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        string msg = <- B;
+        io:println(string `Received string "${msg}" from worker B`);
+    }
+
+    worker B {
+        int num;
+
+        // Receives integer value `10` from worker `A`.
```
Comment (praneesha) [inline examples/async-send/async_send.bal:18]:
```suggestion
        // Receives the `10`  integer value from the `A` worker.
```

File: examples/async-send/async_send.bal [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        num = <- A;
+        io:println(string `Received integer "${num}" from worker A`);
+
+        // Sends string `Hello` to worker `A` asynchronously.
```
Comment (praneesha) [inline examples/async-send/async_send.bal:22]:
```suggestion
        // Sends the `Hello`string to the A` worker ` asynchronously.
```

File: examples/async-send/async_send.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# Async send
+
+Use `-> W` to send a message asynchronously to worker `W`.  These messages should be of `anydata` type. Sending worker will send all messages even though the receiving worker panics or returns an error.
```
Comment (praneesha) [inline examples/async-send/async_send.md:3]:
```suggestion
Use `-> W` to send a message asynchronously to a `W` worker.  These messages should be of the `anydata` type. The sending worker will send all messages even though the receiving worker panics or returns an error.
```

File: examples/async-send/async_send.metatags [hunk 10]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates how async send action can be used for concurrency in Ballerina.
```
Comment (praneesha) [inline examples/async-send/async_send.metatags:1]:
```suggestion
description: This BBE demonstrates how the `async send` action can be used for concurrency in Ballerina.
```

File: examples/sync-send/sync_send.bal [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import ballerina/io;
+
+public function main() {
+    worker A {
+        int msg = 10;
+
+        // Synchronous send to worker `B`. Worker 'A' will wait until 'B' receives the message.
```
Comment (praneesha) [inline examples/sync-send/sync_send.bal:7]:
```suggestion
        // Send a message synchronously to the `B` worker. The  'A' worker will wait until 'B' receives the message.
```

File: examples/sync-send/sync_send.bal [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        error? res = msg ->> B;
+        io:println(res ?: "Transmission to B is successful");
+
+        // This transmission will not happen instead returns `error`.
```
Comment (praneesha) [inline examples/sync-send/sync_send.bal:11]:
```suggestion
        // This transmission will not happen. Instead, it returns an `error`.
```

File: examples/sync-send/sync_send.md [hunk 13]
```diff
@@ -0,0 +1,7 @@
+# Sync send
+
+Use `->> W` to send a message synchronously to worker `W`. These messages should be of `anydata` type. Unlike asynchronous send, synchronous send will wait until the message is delivered. Synchronous send will return nil if the message was successful to the receiver or will return an error or panic based on the receiving worker’s state.
```
Comment (praneesha) [inline examples/sync-send/sync_send.md:3]:
```suggestion
Use `->> W` to send a message synchronously to the `W` worker. These messages should be of the `anydata` type. Unlike asynchronous send, synchronous send will wait until the message is delivered. Synchronous send will return nil if the message was successful to the receiver or will return an error or panic based on the receiving worker’s state.
```

File: examples/sync-send/sync_send.metatags [hunk 14]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates how sync send action can be used for concurrency in Ballerina.
```
Comment (praneesha) [inline examples/sync-send/sync_send.metatags:1]:
```suggestion
description: This BBE demonstrates how the `sync send` action can be used for concurrency in Ballerina.
```

File: examples/sync-send/sync_send.out [hunk 15]
```diff
@@ -0,0 +1,4 @@
+$ bal run sync_send.bal
+Received 10 from worker A
```
Comment (praneesha) [inline examples/sync-send/sync_send.out:2]:
```suggestion
Received 10 from the A worker.
```

File: examples/sync-send/sync_send.out [hunk 16]
```diff
@@ -0,0 +1,4 @@
...
+Transmission to B is successful
+error("Error in worker B")
```
Comment (praneesha) [inline examples/sync-send/sync_send.out:4]:
```suggestion
error("Error in the B worker.")
```

File: examples/sync-send/sync_send.md (hunk 4 above)
Comment (prakanth97) [reply examples/sync-send/sync_send.md:4]: It should be an panic error propagate to the waiting strand. I will update that

**Review by gimantha — APPROVED** [review]

**Pull request merged**